src/components/PeopleTable.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useCallback, useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { PersonRow } from './PersonRow';
+
+type Props = {
+  arrayOfPeople: Child[],
+};
+
+const arrayOfHeaders = [
+  { key: 'Name', label: 'Name' },
+  { key: 'Sex', label: 'Sex' },
+  { key: 'Born', label: 'Born' },
+  { key: 'Died', label: 'Died' },
+  { key: 'Mother', label: 'Mother' },
+  { key: 'Father', label: 'Father' },
+];
+
+export const PeopleTable: React.FC<Props> = React.memo(({ arrayOfPeople }) => {
+  const [sortingOrder, setSortingOrder] = useState<'asc' | 'desc'>('asc');
+  const navigate = useNavigate();
+  const location = useLocation();
+  const searchParams = new URLSearchParams(location.search);
+
+  const sortBy = searchParams.get('sortBy') || '';
+  const sortOrder = searchParams.get('sortOrder') || '';
+
+  const sortTable = (
+    arr: Child[],
+    sort: string | null,
+    order: string | null,
+  ) => {
+    const copy = [...arr];
+
+    switch (sort) {
+      case 'name':
+      case 'sex':
+        return (order === 'asc')
+          ? copy
+            .sort((a, b) => a[sort].localeCompare(b[sort]))
+          : copy
+            .sort((a, b) => b[sort].localeCompare(a[sort]));
```

Comment:
it would be cool to create High order function and pass props in different order to avoid code dublicating